src/main/python/ayab/plugins/ayab_plugin/ayab_control.py

```diff
@@ -415,14 +416,18 @@ def __cnfLine(self, lineNumber):
                     if 0 <= pxlNumber and pxlNumber < 200:
                         self._set_pixel(bytes, pxlNumber)
 
-            # TODO implement CRC8
-            crc8 = 0x00
+            # calculate CRC8 hash
+            hash = crc8.crc8()
+            hash.update(bytes)
+            hash.update(bytearray([reqestedLine]))
```

Comment:
Perhaps you could move this into a separate method which takes the bytes from the whole cnfLine packet and call it in Line 426 and 429. This would enable testing of the method to ensure it yields the same output as in the firmware (e.g. with the same test data as in https://github.com/AllYarnsAreBeautiful/ayab-firmware/blob/v1.0-dev/test/test_serial_encoding.cpp#L49-L55)